tests/tasks/test_binary_container_build.py

```diff
@@ -244,17 +255,31 @@ def mock_build_container(*, build_dir, build_args, dest_tag):
             .once()
             .replace_with(mock_build_container)
         )
+        if not fail_image_size_check:
+            (
+                flexmock(mock_podman_remote)
+                .should_receive("push_container")
+                .with_args(X86_UNIQUE_IMAGE, insecure=REGISTRY_CONFIG["insecure"])
+                .once()
+            )
```

Comment:
Instead of mocking this conditionally, how about
```diff
- once()
+ times(0 if fail_image_size_check else 1)
```
to express that push must not be called if the size check failed

Reply:
that would be better. Updated accordingly